subprojects/docs/src/docs/userguide/repositoryTypes.adoc

```diff
@@ -0,0 +1,390 @@
+[[repository_types]]
+== Repository types
+
+[[sec:flat_dir_resolver]]
+=== Flat directory repository
+
+Some projects might prefer to store dependencies on a shared drive or as part of the project source code instead of a binary repository product. If you want to use a (flat) filesystem directory as a repository, simply type:
+
+++++
+<sample id="flatDirMulti" dir="userguide/artifacts/defineRepository" title="Flat repository resolver">
+    <sourcefile file="build.gradle" snippet="flat-dir-multi"/>
+</sample>
+++++
+
+This adds repositories which look into one or more directories for finding dependencies. Note that this type of repository does not support any meta-data formats like Ivy XML or Maven POM files. Instead, Gradle will dynamically generate a module descriptor (without any dependency information) based on the presence of artifacts. However, as Gradle prefers to use modules whose descriptor has been created from real meta-data rather than being generated, flat directory repositories cannot be used to override artifacts with real meta-data from other repositories. For example, if Gradle finds only `jmxri-1.2.1.jar` in a flat directory repository, but `jmxri-1.2.1.pom` in another repository that supports meta-data, it will use the second repository to provide the module.
+
+For the use case of overriding remote artifacts with local ones consider using an Ivy or Maven repository instead whose URL points to a local directory. If you only work with flat directory repositories you don't need to set all attributes of a dependency. Please see <<sec:optional_attributes>> for more information.
+
+[[sub:maven_central]]
+=== Maven Central repository
+
+Maven Central is a popular repository hosting open source libraries for consumption by Java projects.
+
+To declare the link:https://repo.maven.apache.org/maven2/[central Maven repository] for your build add this to your script:
+
+++++
+<sample id="mavenCentral" dir="userguide/artifacts/defineRepository" title="Adding central Maven repository">
+    <sourcefile file="build.gradle" snippet="maven-central"/>
+</sample>
+++++
+
+[[sub:maven_jcenter]]
+=== Maven JCenter repository
+
+http://bintray.com[Bintray]'s JCenter is an up-to-date collection of all popular Maven OSS artifacts, including artifacts published directly to Bintray.
+
+To declare the link:https://jcenter.bintray.com[JCenter Maven repository] add this to your build script:
+
+++++
+<sample id="mavenJcenter" dir="userguide/artifacts/defineRepository" title="Adding Bintray's JCenter Maven repository">
+    <sourcefile file="build.gradle" snippet="maven-jcenter"/>
+</sample>
+++++
+
+[[sub:maven_google]]
+=== Maven Google repository
```

Comment:
:o: I think it's named "Google Maven Repository" - https://developer.android.com/studio/build/dependencies.html#google-maven

Reply:
fixed